src/renderer/components/output.tsx

```diff
@@ -30,121 +33,195 @@ interface CommandsProps {
-       * binary tree.
-       * @param node A react-mosaic node
-       * @returns Whether that node is a MosaicParent or not
-       */
-      const isParentNode = (
-        node: MosaicNode<WrapperEditorId> | null,
-      ): node is MosaicParent<WrapperEditorId> => {
-        return (node as MosaicParent<WrapperEditorId>)?.direction !== undefined;
-      };
-
-      const { isConsoleShowing } = this.props.appState;
-
-      // this context should always exist, but mocking context in enzyme
-      // is not fully supported, so this condition makes the tests pass
-      if (this.context.mosaicActions) {
-        const mosaicTree = this.context.mosaicActions.getRoot();
-        if (isParentNode(mosaicTree)) {
-          // splitPercentage defines the percentage of space the first panel takes
-          // e.g. 25 would mean the two children panels are split 25%/75%
-          mosaicTree.splitPercentage = isConsoleShowing ? 25 : 0;
-          this.context.mosaicActions.replaceWith([], mosaicTree);
-        }
-      }
+  public async componentDidMount() {
+    autorun(async () => {
+      this.destroyMonaco();
+      await this.initMonaco();
+      this.toggleConsole();
     });
   }
 
+  public componentWillUnmount() {
+    this.destroyMonaco();
+  }
+
   public componentDidUpdate() {
-    if (this.outputRef.current) {
-      this.outputRef.current.scrollTop = this.outputRef.current.scrollHeight;
+    this.toggleConsole();
+  }
+
+  /**
+   * Handle the editor having been mounted. This refers to Monaco's
+   * mount, not React's.
+   */
+  public async editorDidMount(editor: MonacoType.editor.IStandaloneCodeEditor) {
+    const { editorDidMount } = this.props;
+
+    await this.setContent(this.props.appState.output);
+
+    // Notify other editors that the initial editor was mounted
+    if (editorDidMount) {
+      editorDidMount(editor);
     }
   }
 
   /**
-   * Render the timestamp
-   *
-   * @param {number} ts
-   * @returns {string}
+   *  Set Monaco Editor's value.
    */
-  public renderTimestamp(ts: number): string {
-    const { renderTimestamp } = this.props;
+  public async UNSAFE_componentWillReceiveProps(newProps: CommandsProps) {
+    await this.setContent(newProps.appState.output);
+  }
 
-    if (renderTimestamp) {
-      return renderTimestamp(ts);
-    } else {
-      return new Date(ts).toLocaleTimeString();
+  /**
+   * Initialize Monaco.
+   */
+  public async initMonaco() {
+    const { monaco, monacoOptions: monacoOptions } = this.props;
+    const ref = this.outputRef.current;
+    if (ref) {
+      this.setupMonacoLanguage(monaco);
+      this.editor = monaco.editor.create(ref, {
+        language: this.language,
+        theme: 'main',
+        readOnly: true,
+        contextmenu: false,
+        automaticLayout: true,
+        model: null,
+        ...monacoOptions,
+      });
+
+      await this.editorDidMount(this.editor);
     }
   }
 
   /**
+   * Destroy Monaco.
+   */
+  public destroyMonaco() {
```

Comment:
_nit_ I'd have a slight preference to a name like `destroyMonacoEditor()` or `destroyEditor()` rather than `destroyMonaco()` because of the strange hoops we jump through to load the `monaco` object, so it's slightly clearer to not imply that we're destroying that `monaco` object

Reply:
Oops, I didn't realize this was inherited from the older code! :dizzy_face: